oxidation/bindings/generator/api.py

```diff
@@ -25,35 +25,63 @@ class Ref(Generic[TypeVar("REFERENCED")]):  # noqa
 
 # A type that should be converted from/into string
 class StrBasedType:
-    pass
+    ...
 
 
 class OrganizationID(StrBasedType):
-    pass
+    ...
+
+
+class DeviceLabel(StrBasedType):
+    ...
+
+
+class HumanHandle(StrBasedType):
+    ...
+
+
+class FSPath(StrBasedType):
+    ...
 
 
 class DeviceID(StrBasedType):
-    pass
+    ...
 
 
-class DeviceLabel(StrBasedType):
-    pass
+class DeviceFileType(Variant):
+    class Password:
+        ...
 
+    class Recovery:
+        ...
 
-class PathBuf(StrBasedType):
-    pass
+    class Smartcard:
+        ...
+
+
+class LocalDeviceError(Variant):
```

Comment:
Do we need all those errors types ?

I even wonder if we need error at all for `list_available_devices` 🤔 

The idea is to provide a convenient API for the GUI to use, hence providing many errors is often the mark we are exposing internal details and making the API more complicated than it should be.

On top of that the API between GUI and libparsec don't have any backward compatibility requirements, so we can start simple and add errors later on when the need come ^^

Reply:
@Max-7 @NicoTuxx Any suggestion here ?

Reply:
Done